_api-reference/ingest-apis/ip2geo.md

```diff
@@ -0,0 +1,212 @@
+
+#### Example: JSON response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Sending a GET request
+
+To get information about one or more IP2geo datasource, send a GET request.  
+
+#### Example: Request
+
+```
+GET /_plugins/geospatial/ip2geo/datasource/my-datasource
+```
+
+#### Example: Response
+
+```
+{
+  "datasources": [
+    {
+      "name": "my-datasource",
+      "state": "AVAILABLE",
+      "endpoint": "https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json",
+      "update_interval_in_days": 3,
+      "next_update_at_in_epoch_millis": 1685125612373,
+      "database": {
+        "provider": "maxmind",
+        "sha256_hash": "0SmTZgtTRjWa5lXR+XFCqrZcT495jL5XUcJlpMj0uEA=",
+        "updated_at_in_epoch_millis": 1684429230000,
+        "valid_for_in_days": 30,
+        "fields": [
+          "country_iso_code",
+          "country_name",
+          "continent_name",
+          "region_iso_code",
+          "region_name",
+          "city_name",
+          "time_zone",
+          "location"
+        ]
+      },
+      "update_stats": {
+        "last_succeeded_at_in_epoch_millis": 1684866730192,
+        "last_processing_time_in_millis": 317640,
+        "last_failed_at_in_epoch_millis": 1684866730492,
+        "last_skipped_at_in_epoch_millis": 1684866730292
+      }
+    }
+  ]
+}
+```
+
+## Updating IP2geo datasource
+
+Update IP2geo datasource. The GeoIP data from the new endpoint should contain all fields in GeoIP data from the current endpoint for the update to succeed.
+
+#### Example: Request
+
+```json
+{
+    "endpoint": https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json,
+    "update_interval_in_days": 10
+}
+```
+
+#### Example: Response
+
+```json
+{
+    "acknowledged":true
+}
+```
+
+## Creating the IP2geo processor
+
+
+
+#### Example: Request
+
+```json
+{
+   "description":"convert ip to geo",
+   "processors":[
+    {
+        "ip2geo":{
+            "field":"ip",
+            "datasource":"my-datasource"
+        }
+    }
+   ] 
+}
+
+#### Example: Response
+
+```json
+{
+	"acknowledged": true
+}
+```
+
+## Using the IP2geo processor in a pipeline
+
+The following table describes `ip2geo` options
+
+| Name | Required | Default | Description |
+|------|----------|---------|-------------|
+| field | yes | - | The field to get the ip address from for the geographical lookup. |
+| target_field | no | ip2geo | The field that will hold the geographical information looked up from the Maxmind database. |
+| database | no | geoip.maps.opensearch.org/v1/geolite2-city/manifest.json | The database filename referring to a database the module ships with or a custom database in the ingest-geoip config directory. |
```

Comment:
```suggestion
| datasource | yes | - | The datasource name to look up geographical information |
```

Reply:
Suggestion means, replace the line with suggested one. I can still see "database" which should not be there.